pkg/logql/bench/store_dataobj_v2_engine.go

```diff
@@ -38,31 +46,119 @@ func NewDataObjV2EngineStore(dir string, tenantID string) (*DataObjV2EngineStore
 	})
 }
 
-func dataobjV2StoreWithOpts(dataDir string, tenantID string, cfg engine.Config, metastoreCfg metastore.Config) (*DataObjV2EngineStore, error) {
+func dataobjV2StoreWithOpts(dataDir string, tenantID string, cfg engine.ExecutorConfig, metastoreCfg metastore.Config) (*DataObjV2EngineStore, error) {
 	logger := log.NewNopLogger()
 
-	// Setup filesystem client as objstore.Bucket
-	// This assumes the engine is configured to read its specific data format (e.g., Parquet)
-	// from this directory structure. The existing benchmark data generation might need adjustments
-	// if it doesn't produce this format.
-	// Use NewBucket from thanos-io/objstore/providers/filesystem, similar to store_dataobj.go
+	if testing.Verbose() {
+		startTime := time.Now()
+
+		logger = log.NewLogfmtLogger(&unescapeWriter{w: log.NewSyncWriter(os.Stderr)})
+
+		// Rather than reporting timestamp, it can be useful in the context of
+		// our tests and benchmarks to report the elapsed time since the store
+		// was created.
+		logger = log.With(logger, "elapsed", log.Valuer(func() any {
+			return time.Since(startTime).String()
+		}))
+	}
+
 	storeDir := filepath.Join(dataDir, "dataobj")
 	bucketClient, err := filesystem.NewBucket(storeDir)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create filesystem bucket for DataObjV2EngineStore: %w", err)
 	}
 
-	// Instantiate the new engine
-	// Note: The tenantID is not directly passed to engine.New here.
-	// The engine might expect tenant information to be part of the query context
-	// or derived from the bucket structure if it's multi-tenant aware.
-	// This might require adjustment based on how pkg/engine/engine actually handles multi-tenancy
-	// with a generic objstore.Bucket.
-	queryEngine := engine.New(cfg, metastoreCfg, bucketClient, logql.NoLimits, nil, logger)
+	sched := &engine.Scheduler{Logger: logger}
+	schedService, err := sched.Service()
+	if err != nil {
+		return nil, fmt.Errorf("creating scheduler service: %w", err)
+	} else if err := schedService.StartAsync(context.Background()); err != nil {
+		return nil, fmt.Errorf("starting scheduler service: %w", err)
+	} else if err := schedService.AwaitRunning(context.Background()); err != nil {
+		return nil, fmt.Errorf("awaiting scheduler service to run: %w", err)
+	}
```

Comment:
The `services` package provides a utility:

```suggestion
	if err := services.StartAndAwaitService(context.Background(), schedService); err != nil {
		return nil, fmt.Errorf("starting scheduler service: %w", err)
	}
```